geode-management/src/main/java/org/apache/geode/management/api/ClusterManagementOperationResult.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.management.api;
+
+import java.util.concurrent.CompletableFuture;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+import org.apache.geode.annotations.Experimental;
+
+@Experimental
+public class ClusterManagementOperationResult<V extends JsonSerializable>
```

Comment:
This looks like a class that should instead be an interface. I don't think we want users of the java api to ever create instances of it. The problem is ClusterManagementResult is a class that should also be an interface.

Reply:
There's not much benefit of making an interface here, it's all internal and Jackson needs some concrete class to deserialize to.

Reply:
Let's explore this idea in a separate PR.  Interfaces are a good practice for "future-proofing" and might be worth considering across the Cluster Management API